Smile/src/main/java/smile/validation/ConfusionMatrix.java

```diff
@@ -0,0 +1,60 @@
+package smile.validation;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Generates the confusion matrix based on truth and prediction vectors
+ * @author owlmsj
+ *
+ */
+public class ConfusionMatrix {
+	
+	private int[][] matrix;
+	
+	
+	public int[][] getMatrix(){
+		return matrix;
+	}
+	
+	public int[][] generate(int[] truth, int[] prediction) {
```

Comment:
Nice work. Shall we just change generate to the constructor. It is not necessary to create an object and then call this method. Let's do it in one step. Thanks!